src/storage/btree.c

```diff
@@ -14329,8 +14332,35 @@ btree_find_rightmost_leaf (THREAD_ENTRY * thread_p, BTID * btid, VPID * pg_vpid,
   return btree_find_boundary_leaf (thread_p, btid, pg_vpid, stat_info, BTREE_BOUNDARY_LAST);
 }
 
+static PAGE_PTR
+btree_find_boundary_leaf (THREAD_ENTRY * thread_p, BTID * btid, VPID * pg_vpid, BTREE_STATS * stat_info,
+			  BTREE_BOUNDARY where)
+{
+  PAGE_PTR page = nullptr;
+  bool desync_occured = false;
+
+  do
+    {
+      page =
+	btree_find_boundary_leaf_with_repl_desync_check (thread_p, btid, pg_vpid, stat_info, where, &desync_occured);
+#ifdef SERVER_MODE
+      if (desync_occured)
+	{
+	  //wait for replication
+	  passive_tran_server *const pts_ptr = get_passive_tran_server_ptr ();
+	  const LOG_TDES *const tdes = LOG_FIND_CURRENT_TDES (thread_p);
+	  assert (tdes->page_desync_lsa.is_null ());
```

Comment:
```suggestion
	  assert (!tdes->page_desync_lsa.is_null ());
```
LSA is set downwards two levels in `pgbuf_fix_old_and_check_repl_desync`